core/io/resource.cpp

```diff
@@ -200,7 +266,10 @@ Ref<Resource> Resource::duplicate_for_local_scene(Node *p_for_scene, HashMap<Ref
 		if (!(E.usage & PROPERTY_USAGE_STORAGE)) {
 			continue;
 		}
-		Variant p = get(E.name);
+		Variant p = get(E.name).duplicate(true);
+
+		_dupe_sub_resources(p, p_for_scene, remap_cache);
```

Comment:
You can move the code below to `_dupe_sub_resources()`.

Reply:
Good point! I'm not deep into C++ so I thought doing `p = ...` wouldn't work from inside a function.
But references seem to allow this :D